pkg/network/telemetry/utils.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/common"
+)
+
+func contains(want string, strings []string) bool {
+	for _, candidate := range strings {
+		if want == candidate {
+			return true
+		}
+	}
+	return false
+}
+
+// Example: given {"_opt3", "tag:a", "_opt2", "tag:b", "_opt1", "tag:c"}
+// The return value will be:
+// {"tag:a", "tag:b", "tag:c"}
+// {"_opt1", "_opt2", "_opt3"}
+func splitTagsAndOptions(all []string) (tags, opts []string) {
+	if len(all) == 0 {
+		return
+	}
+	tagSet := common.NewStringSet()
+	optSet := common.NewStringSet()
+
+	for _, s := range all {
+		if strings.HasPrefix(s, optPrefix) {
+			optSet.Add(s)
+		} else {
+			tagSet.Add(s)
+		}
+	}
+
+	tags = tagSet.GetAll()
+	opts = optSet.GetAll()
+	sort.Strings(tags)
+	sort.Strings(opts)
```

Comment:
why sorting?

Reply:
to make it deterministic and easier to compare:
https://github.com/DataDog/datadog-agent/blob/4b369b57d1fcd374f4655ad2d2369340e21e8b58/pkg/network/telemetry/metric.go#L112-L125
I can leave a comment in the code to clarify

Reply:
a comment will be great! 
if you have some internal assumptions in one file and you are relying on it in another file -> it should be documented for the other developers